src/woo_search/search_index/api/viewsets/search.py

```diff
@@ -0,0 +1,112 @@
+from django.utils.translation import gettext_lazy as _
+
+from drf_spectacular.utils import extend_schema
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from woo_search.search_index.api.serializers import (
+    SearchResponseSerializer,
+    SearchSerializer,
+)
+
+from ...client import get_client
+from ...constants import SortChoices
+from ...typing import SearchType
+
+
+@extend_schema(
+    tags=["index"],
+    summary=_("Search"),
+    description=_("Search the publication and document records."),
+    responses={
+        200: SearchResponseSerializer(many=True),
+    },
+)
+class SearchView(APIView):
+    serializer_class = SearchSerializer
+    response_serializer_class = SearchResponseSerializer
+
+    def _query_multi_match(self, query: str):
+        return {
+            "multi_match": {
+                "fields": [
+                    "officiele_titel^2",
+                    "verkorte_titel^1.5",
+                    "omschrijving",
+                    "identifier^3",
+                ],
+                "fuzziness": 1,
+                "query": query,
+            }
+        }
+
+    def _build_must(self, params: SearchType):
+        must = []
+
+        if query := params.get("query"):
+            must.append(self._query_multi_match(query))
+
+        return must
+
+    def base_search_query(self, params: SearchType):
```

Comment:
this is also an unnecessary method IMO. It's really hard for me to follow the flow of calls.